### ISSUE TITLE
🐛 Bug Report: tasks doesn't get deleted if plugin gets removed

### ISSUE THREAD

**Issue by LegendSebastianoL** [backstage/backstage#25511]
### 📜 Description

I noticed that if you remove a plugin that has a tasks scheduler once you inspect the Database the task is still there, there is no task cleanup. This also happens if you create a custom entity provider and you change the `getProviderName` return, a new task is created but the old one is still there.

### 👍 Expected behavior

if you remove a plugin with a task, this one should be removed from the Database at some point.

### 👎 Actual Behavior with Screenshots

if you remove a plugin and connect to the database, even after a long time (months), you have old tasks connected to a plugin that doesn't exist anymore.

### 👟 Reproduction steps

- install a plugin with a scheduler
- start backstage so that the task get inserted in the DB
- uninstall the plugin
- start backstage again and all tasks are there, you can even leave it there for weeks but the task never disappears.

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

None

**Comment by freben** [comment]
Right, this was actually intentionally left out in the initial implementation and never prioritized to add in. So it's a known harmless omission.

The dataset is typically microscopic, and having entries there does not mean that work is really being performed. It's only a state bookkeeping dataset. So the only actual waste is the few bytes of storage, and potential confusion for those who look at the tables.

It's not entirely trivial to know when to garbage collect a row, in an imperfect world. Would probably need to amend the scheduler with some form of heartbeat mechanism that keeps rows alive as long as any host exists that is interested in the task at hand.

**Comment by sebalaini** [comment]
I never dug into how the scheduler works and whether having those "useless" tasks saved in the DB is an issue or not.

For me as long as it doesn't create any issues with existing/renamed tasks it's not a problem :) Should I close the task or is it something that's in your roadmap to add as a feature?

**Comment by freben** [comment]
Yep I think from a runtime point of view, all is fine, and in that regard it's OK to close if you're alright with that. Feel free to open a cleanup feature request issue if this does cause problems in the future - but hopefully it won't. At least not until we add introspection of the scheduler state, at which point we may have to deal with this to ensure that the admin views are clean of cruft 😅

**Comment by LegendSebastianoL** [comment]
OK I'll close the issue then :)
